include/wx/aui/auibar.h

```diff
@@ -494,17 +514,29 @@ class WXDLLIMPEXP_AUI wxAuiToolBar : public wxControl
     bool SetFont(const wxFont& font) override;
 
 
+    wxAuiToolBarItem* AddTool(wxAuiToolBarItem* item);
+
     wxAuiToolBarItem* AddTool(int toolId,
                  const wxString& label,
                  const wxBitmapBundle& bitmap,
                  const wxString& shortHelpString = wxEmptyString,
-                 wxItemKind kind = wxITEM_NORMAL);
+                 int kind = wxAUI_TB_ITEM_NORMAL)
```

Comment:
Why `int` and not `wxAuiToolBarItemKind`?

Reply:
`int` is used to preserve compatibility with code using the old `wxITEM_*` identifiers here. Changing the type of the argument to `wxAuiToolBarItemKind` forces the use of `wxAUI_TB_ITEM_*` constants and causes a compile error when any `wxITEM_*` constants are used, because they can't be implicitly converted between the enums. Both enums can be implicitly converted to `int` though, so changing the `AddTool` functions to take `int` allows it to accept both the new `wxAUI_TB_ITEM_*` constants and the old `wxITEM_*` constants that existing code is using. (and note that `wxAuiToolBarItem` already treats kind as an `int`` in its API and storage, so this is the only change needed to make it work).

Reply:
Yes, but this is really a pity as it's not type-safe at all. Could we provide both overloads, taking the old and the new enum? It's annoying, of course, but I think this would be worth doing.